src/solver/solver.cxx

```diff
@@ -468,8 +468,28 @@ int Solver::solve(int NOUT, BoutReal TIMESTEP) {
     throw BoutException(_("Failed to initialise solver-> Aborting\n"));
   }
 
+  // Set the run ID to a random number
+  run_restart_from = run_id; // Restarting from the previous run ID
+
+  if (MYPE == 0) {
+    srand(static_cast<unsigned int>(time(nullptr)));
```

Comment:
If I'm reading right, `time()` returns output in seconds. If a script started lots of runs in a scan in parallel, you might get a clash. Maybe we could use a high resolution clock http://www.cplusplus.com/reference/chrono/high_resolution_clock/now/? To be really paranoid we could also combine with a PID, multiplying by some long-ish amount of time so different PIDs aren't the same as small changes in time, something like
```
srand(static_cast<unsigned int>(std::chrono::high_resolution_clock::now()) + 1000000000000u * unsigned_int(get_pid()));
```
maybe?

Reply:
We should probably at least use the C++ `<random>`:

```cpp
    std::random_device rd;  // cross platform way to get a random seed
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> dis(2, 255); // ensures it's never 0 or 1, probably want std::numeric_limits<init>::max()
    run_id = dis(gen);
```

We might like to go one step further and generate something like a UUID, there are a few header-only libraries out there to do this.